syntax = "proto2";

package zjchain.zbft.protobuf;
import "protos/block.proto";
import "protos/pools.proto";

message TxBft {
    optional block.protobuf.Block block = 1;
    optional bytes prepare_final_hash = 2;
    optional uint64 height = 3;
    repeated uint64 bloom_filter = 4;
    repeated bytes tx_hash_list = 5;
    optional pools.protobuf.StepType tx_type = 6 [default=kNormalFrom];
}

message ZbftMessage {
    optional bytes prepare_gid = 1;
    optional bytes precommit_gid = 2;
    optional bytes commit_gid = 3;
    optional bool leader = 4;
    optional uint32 net_id = 5;
    repeated uint64 bitmap = 6;
    optional bool agree_prepare = 7 [default=false];
    optional bool agree_precommit = 8 [default=false];
    optional bool agree_commit = 9 [default=false];
    optional uint32 pool_index = 10;
    optional int32 error = 11;
    optional TxBft tx_bft = 12;
    optional uint32 member_index = 13;
    optional bytes backup_enc_data = 14;
    optional uint64 elect_height = 15;
    optional bytes bls_sign_x = 16;
    optional bytes bls_sign_y = 17;
    repeated uint64 commit_bitmap = 18;
    optional bytes prepare_hash = 19;
}
